pkg/bgpv1/types/bgp.go

```diff
@@ -46,6 +46,23 @@ type NeighborRequest struct {
 	VR       *v2alpha1api.CiliumBGPVirtualRouter
 }
 
+// SoftResetDirection defines the direction in which a BGP soft reset should be performed
+type SoftResetDirection int
+
+const (
+	SoftResetDirectionIn SoftResetDirection = iota
+	SoftResetDirectionOut
+	SoftResetDirectionBoth
+)
+
+// ResetNeighborRequest contains parameters used when resetting a BGP peer
+type ResetNeighborRequest struct {
+	PeerAddress        string
+	Soft               bool
+	SoftResetDirection SoftResetDirection
```

Comment:
Instead of exposing a `bool`, could `SoftResetDirection` be a pointer, and if `nil`, soft reset would be considered `false`? This can be resolved in a follow-up PR if you agree.

Reply:
I am not sure here, I feel that being explicit sometimes may be better for readability - and pointer type requires an assignment to a var upon initialization to use a constant value, so the API is not that easy to use... but yeah, if you feel like it would be better, we can try that in a separate PR.